GRDB/Core/Support/Foundation/Data.swift

```diff
@@ -25,6 +25,17 @@ extension Data : DatabaseValueConvertible, StatementColumnConvertible {
     /// a Blob.
     public static func fromDatabaseValue(_ dbValue: DatabaseValue) -> Data? {
         guard case .blob(let data) = dbValue.storage else {
+            // Check to see if data is String, if so then pass is through JSONSerialization
+            // to confirm if contains JSON, if so then this is a nested type stored as JSON,
+            // so return string as Data so that it can be decoded
```

Comment:
I'm sorry @gusrota, but we have a deep misunderstanding. I'm sure it's my fault. JSON has nothing to do here. Conversion from String to Data should always succeed:

```swift
switch dbValue.storage {
    case .blob(let data): return data
    case .string(let string): return string.data(using: .utf8)
    default: return nil
}
```

I'm also concerned by the fact that you don't hesitate decoding JSON data multiple times without any concern on performance.

Reply:
When we try this code we get 7 tests failing 
<img width="629" alt="screen shot 2018-08-14 at 13 26 56" src="https://user-images.githubusercontent.com/108781/44091604-eb2ae000-9fc5-11e8-9229-aece8e00534a.png">
should the tests be fixed or should we handle strings in some other way so the tests pass ?

Reply:
Yes, that test needs to be fixed 😄 (not by reversing the nil/not nil test, but by checking the converted value, of course)